### PR TITLE
e2e: add API call budget check

### PR DESCRIPTION
**Pull request by sjenning**
**What this PR does / why we need it**:
Now, that we've done the work to root out causes of high API activity, lets seal against further regression.

This introduces a check to ensure that the CPO and HO don't go over a certain read/mutate budget

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: test/e2e/util/util.go [hunk 1]
```diff
@@ -283,3 +290,144 @@ func EnsureNodeCountMatchesNodePoolReplicas(t *testing.T, ctx context.Context, h
...
+				budget: 1500,
+			},
+			{
+				name:   "hypershift-operator no 404 deletes",
+				query:  "sum(hypershift:operator:component_api_requests_total{method=\"DELETE\", code=\"404\"})",
+				budget: 0,
+			},
+		}
+
+		for _, budget := range budgets {
```
Comment (alvaroaleman) [inline test/e2e/util/util.go:407]: Maybe make these subtests `(t.Run(budged.name, func(t *testing.T){testcode})` so this gets more legible in the error case?

File: test/e2e/util/util.go [hunk 2]
```diff
@@ -283,3 +290,144 @@ func EnsureNodeCountMatchesNodePoolReplicas(t *testing.T, ctx context.Context, h
...
+		promRoute := &routev1.Route{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "thanos-querier",
+				Namespace: "openshift-monitoring",
+			},
+		}
+		if err := client.Get(ctx, crclient.ObjectKeyFromObject(promRoute), promRoute); err != nil {
+			t.Skip("unable to get prometheus route, skipping")
+		}
+		promEndpoint := fmt.Sprintf("https://%s", promRoute.Status.Ingress[0].Host)
```
Comment (alvaroaleman) [inline test/e2e/util/util.go:341]: Shouldn't happen but maybe lencheck this first so we don't cause an out of bounds if for some reason the route doesn't have a host?

**Comment by sjenning** [comment]
/hold
wanting to address review feedback and incorporate ability gained from https://github.com/openshift/hypershift/pull/1036

**Review by alvaroaleman — APPROVED** [review]

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**